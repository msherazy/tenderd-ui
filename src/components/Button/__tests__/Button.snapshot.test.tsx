import { render } from '@testing-library/react';
import { Index as Button } from '../index';

describe('Button Component Snapshots', () => {
  test('primary button renders correctly', () => {
    const { container } = render(<Button variant="primary">Primary Button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('secondary button renders correctly', () => {
    const { container } = render(<Button variant="secondary">Secondary Button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('danger button renders correctly', () => {
    const { container } = render(<Button variant="danger">Danger Button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('ghost button renders correctly', () => {
    const { container } = render(<Button variant="ghost">Ghost Button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('full width button renders correctly', () => {
    const { container } = render(<Button fullWidth>Full Width Button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('loading button renders correctly', () => {
    const { container } = render(<Button isLoading>Loading Button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('disabled button renders correctly', () => {
    const { container } = render(<Button disabled>Disabled Button</Button>);
    expect(container).toMatchSnapshot();
  });
});
