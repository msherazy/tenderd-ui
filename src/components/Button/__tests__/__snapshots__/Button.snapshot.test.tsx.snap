// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshots danger button renders correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto bg-red-600 text-white hover:bg-red-700 focus:ring-2 focus:ring-offset-2 focus:ring-red-500"
  >
    Danger Button
  </button>
</div>
`;

exports[`Button Component Snapshots disabled button renders correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto hover:opacity-90 focus:ring-2 focus:ring-offset-2 focus:ring-primary"
    disabled=""
    style="color: rgb(255, 255, 255);"
  >
    Disabled Button
  </button>
</div>
`;

exports[`Button Component Snapshots full width button renders correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-full hover:opacity-90 focus:ring-2 focus:ring-offset-2 focus:ring-primary"
    style="color: rgb(255, 255, 255);"
  >
    Full Width Button
  </button>
</div>
`;

exports[`Button Component Snapshots ghost button renders correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto text-gray-500 hover:text-primary hover:border-primary focus:text-primary focus:border-primary"
  >
    Ghost Button
  </button>
</div>
`;

exports[`Button Component Snapshots loading button renders correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto hover:opacity-90 focus:ring-2 focus:ring-offset-2 focus:ring-primary"
    disabled=""
    style="color: rgb(255, 255, 255);"
  >
    <span
      class="w-4 h-4 mr-2 border-2 border-t-transparent border-current rounded-full animate-spin"
    />
    Loading Button
  </button>
</div>
`;

exports[`Button Component Snapshots primary button renders correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto hover:opacity-90 focus:ring-2 focus:ring-offset-2 focus:ring-primary"
    style="color: rgb(255, 255, 255);"
  >
    Primary Button
  </button>
</div>
`;

exports[`Button Component Snapshots secondary button renders correctly 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto bg-gray-200 text-gray-800 hover:bg-gray-300 focus:ring-2 focus:ring-offset-2 focus:ring-gray-300"
  >
    Secondary Button
  </button>
</div>
`;
